### ISSUE TITLE
Google Meet "You can't join this video call" when not logged in (public meeting)

### ISSUE THREAD

**Issue by emmahsax** [brave/brave-browser#32836]
## Description 

When not logged into Google, if I try to navigate to a running Google Meet that is open to the public, Brave gives me this message:

![brave denied clean](https://github.com/brave/brave-browser/assets/7562793/effa06e9-a47d-4ba0-b856-9dd565810347)

However, if I log into any Google account (it doesn't have to be the one that created the Google Meet), then it lets me join.

Here's the settings for the Google Meet:

<img width="358" alt="settings" src="https://github.com/brave/brave-browser/assets/7562793/1a12b6cd-61b3-40ea-bf4d-472ba2a8a793">

I've tested that I can join that public Google Meet from other browsers such as Google Chrome, Safari, and Firefox without being logged into Google.

## Steps to Reproduce

   1. Create an instant Google Meet with your Google account
   2. Go into the settings bar on the bottom right corner and switch that Google Meet to public
   3. Create an Incognito/Private window **OR** a new Brave profile where you're not logged into Google
   4. Navigate to the same Google Meet URL you started in Step 1
   5. Denied!

In my testing, it didn't matter whether the Admin (Google user who created the Google Meet) was in the meeting or not (in the first browser window)... Brave still wouldn't let me through if the profile I was trying to work with was not logged in (in the second browser window OR Incognito). But, if you run into trouble reproducing, then perhaps try having the Admin leave the meeting after Step 2 in the reproduction steps.

## Actual result:

See first screenshot attached.

## Expected result:

I am allowed to join the Google Meet.

## Reproduces how often: 

Easily reproduced today... first day running into the issue, but also the first day I've tried this workflow, so the issue probably existed before today.

### Brave

`Version 1.57.62 Chromium: 116.0.5845.180 (Official Build) (arm64)`

### MacOS

`MacOS Ventura 13.5.1 (22G90)`

## Version/Channel Information:

- Can you reproduce this issue with the current release?
  - Yes (with version listed above)
- Can you reproduce this issue with the beta channel?
  - Yes (with version `Version 1.58.109 Chromium: 116.0.5845.163 (Official Build) beta (arm64)`)
- Can you reproduce this issue with the nightly channel? 
  - Yes (with version `Version 1.59.76 Chromium: 116.0.5845.163 (Official Build) nightly (arm64)`)

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
  - No
- Does the issue resolve itself when disabling Brave Rewards?
  - No
- Is the issue reproducible on the latest version of Chrome?
  - No (with version `Version 116.0.5845.179 (Official Build) (arm64)`)

**Comment by rebron** [comment]
cc: @ryanbr

**Comment by kanishk6103** [comment]
Same issue! it's very annoying, lately brave has been giving me a headache

**Comment by Emi-TheDhamphirInLoveUnderTheFrozenStar** [comment]
In case someone knows, Fingerprinting protections needs to be set to `Allow Fingerprinting` for Google Meet to work.
So it is just another case of Google not wanting _Brave to add subtle noise to APIs commonly used for fingerprinting_, so it throws an error which can be seen in Console, complaining about errors and causes for it.
So Google just decided to change something recently to throw an error With Brave Fingerprinting protections.

**Comment by Roko131** [comment]
Same with `v1.58.129`, in incognito.

It works when `shields` are down.

Console errors:
![image](https://github.com/brave/brave-browser/assets/3532929/76ac5908-e194-4c97-8a8a-cd386ed9f8de)

**Comment by Roko131** [comment]
> Interesting. I thought it was still broken with Shields down, but maybe I don't know how to disable Shields.

To disable shields:
1. click the `brave shield` icon on top
2. Toggle "Shields are UP"

![image](https://github.com/brave/brave-browser/assets/3532929/9eb5b5bf-ca13-4082-b9db-1a1f7611aaa7)

**Comment by ryanbr** [comment]
Which changing sheilds, I setup a meet call. Can't share the 1 video camera, but disabling the camera on the host and allowing to join using Brave worked for me. I was signed into Google, try signing in using Google first.

**Comment by mvl22** [comment]
> It is due to 'Block Fingerprinting', disable that.

That seems to be a global setting, though. It feels less problematic to put shields down just for this single site.

**Comment by mvl22** [comment]
Ah, I see it now. It's in the 'Advanced controls' expandable area.

![Image](https://github.com/user-attachments/assets/2f5341c3-3371-489e-bdb0-5c42cfecdd85)

**Comment by benji1000** [comment]
Hello, I experienced the same issue and resolved it using Roko131 and IkelAtomig advices. Thank you both!

I'm bumping this ticket to spark a discussion on how to prevent or mitigate this issue, as I'm fairly certain that we are not just a handful of people in this conversation experiencing it. Having such a popular website not working on the first go in Brave, but working in other browsers (which may be the first reflex of users when a website doesn't work in a browser), may harm Brave's reputation, which could eventually be seen as "the browser where websites don't work". I guess we don't want that.

I understand that privacy is one of the goals of Brave, and that browser fingerprinting is a mechanism that can reduce privacy. However, it asks a question about the balance between privacy and usability, and Brave's adoption among users.

This question can easily be dismissed by saying that Brave users should be aware of potentially broken websites due to Brave's privacy-first nature, and that they should disable the shield. If that's the case, this ticket can be closed right now, no need to keep it open! But it nevertheless may confuse new users, or even long-term Brave users who happen to never have experienced a broken website, and may not be familiar with Brave's shield and the possibility to disable it. Therefore, reducing Brave's adoption.

That's why I'm wondering if there could be a better solution to preserve Brave's privacy-first nature, while ensuring a smooth experience for users. Should Google Meet, and potentially a list of other rather safe and popular websites (after some kind of review or categorization) be completely exempted from the anti-fingerprinting mechanism? Should Brave ask the user, when creating a new profile, if they want to have the full, privacy-first experience (as of right now), or if they want a balanced experience (with a list of acceptable websites where privacy is slightly lowered to better accommodate usability)? Should Brave display some kind of popup to the user when browsing these websites, to say that the website they are on doesn't work well with one of Brave's privacy mechanisms, and suggest/offer to disable it? These are just the ideas I'm thinking of.

**Comment by Ufira** [comment]
<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/d7bbe984-1571-48c5-b366-d626abbc872e" />

<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/890b1f4a-8eaf-4587-a3b2-4494a908fea3" />

<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/b168143f-8fb2-4738-ae2e-71eec5ce8dd4" />

Disabling Fingerprint -> Canvas, did the trick

**Comment by bsclifton** [comment]
@ShivanKaul what's a good next step here?  I saw https://github.com/brave/adblock-lists/pull/2648 was merged; is this good to close?

**Comment by ShivanKaul** [comment]
@ryanbr can you confirm if this issue can be closed?

**Comment by ryanbr** [comment]
yeah, lets close. The fix landed. If its still an issue we re-open if needed.

**Comment by IkelAtomig** [comment]
You need to reopen this, now along with canvas, webgl also needs to be made and exception to work.
